exchange/bitswap/decision/peer_request_queue.go

```diff
@@ -198,12 +266,20 @@ type activePartner struct {
 
 	// priority queue of tasks belonging to this peer
 	taskQueue pq.PQ
+
+	//  kinda dirty to give each active partner all the ledgers and it's pid
+	//  instead of just its own ledger, but i'm not sure how else to deal with
+	//  the lazily loaded ledgers
```

Comment:
oh by "giving each active partner all the ledgers" that i just meant i thought it was weird to give an active partner a pointer to a map of ledgers instead of just a pointer to its own ledger.  maps are passed by reference right?  so aren't state updates still happening in one place?  sorry, i'm still kinda confused